frontend/packages/console-app/src/components/cloud-shell/CloudShellTerminal.tsx

```diff
@@ -102,17 +88,25 @@ const CloudShellTerminal: React.FC<CloudShellTerminalProps> = ({ user, onCancel
     };
   }, [username, workspaceName, workspaceNamespace, workspacePhase]);
 
+  // failed to load the workspace
   if (loadError) {
     return (
       <StatusBox loaded={loaded} loadError={loadError} label="OpenShift command line terminal" />
     );
   }
 
+  // failed to init the terminal
   if (initError) {
     return <LoadError message={initError} label="OpenShift command line terminal" />;
   }
 
-  if (!loaded || (workspaceName && !initData)) {
+  // loading the workspace resource
+  if (!loaded) {
```

Comment:
nit: Did we need to split it? It could've been `<TerminalLoadingBox message={!loaded?"":undefined}  />`

Reply:
Yes it must be split because the other one is wrapped in another div for the background.